manager/controllers/app/fybrikapplication_controller.go

```diff
@@ -798,11 +798,40 @@ func (r *FybrikApplicationReconciler) GetAllModules() (map[string]*fappv1.Fybrik
 		return moduleMap, err
 	}
 	for ind := range moduleList.Items {
+		if !validateModule(&moduleList.Items[ind]) {
+			continue
+		}
 		moduleMap[moduleList.Items[ind].Name] = &moduleList.Items[ind]
 	}
 	return moduleMap, nil
 }
 
+// validate that the module can be selected:
+// 1. check the module validation status
+// 2. special processing for modules that support any connections from a specific catalog provider -
+// set the protocol value to an empty one (empty values stand for "*" and are not checked)
+func validateModule(module *fappv1.FybrikModule) bool {
+	if len(module.Status.Conditions) > 0 &&
+		module.Status.Conditions[ModuleValidationConditionIndex].Status == v1.ConditionFalse {
+		return false
+	}
+	for capabilityInd := range module.Spec.Capabilities {
+		for interfaceInd := range module.Spec.Capabilities[capabilityInd].SupportedInterfaces {
+			if module.Spec.Capabilities[capabilityInd].SupportedInterfaces[interfaceInd].Source != nil &&
+				module.Spec.Capabilities[capabilityInd].SupportedInterfaces[interfaceInd].Source.Protocol ==
+					taxonomy.ConnectionType(environment.GetCatalogProvider()) {
+				module.Spec.Capabilities[capabilityInd].SupportedInterfaces[interfaceInd].Source.Protocol = ""
+			}
+			if module.Spec.Capabilities[capabilityInd].SupportedInterfaces[interfaceInd].Sink != nil &&
+				module.Spec.Capabilities[capabilityInd].SupportedInterfaces[interfaceInd].Sink.Protocol ==
+					taxonomy.ConnectionType(environment.GetCatalogProvider()) {
```

Comment:
Should the string be converted to lower case before comparing?

Reply:
Added insensitive comparison.